templates/nginx.conf.j2

```diff
@@ -97,38 +99,35 @@ http {
             fastcgi_buffers {{served_domain.fastcgi_buffers}}
             {% endif %}
 
-            {% if served_domain.configurations is defined%}
-                {% for configuration in served_domain.configurations %}
-                    {{ configuration.content }}
-                {% endfor %}
-            {% endif %}
-            
+            {% for configuration in served_domain.configurations | default([]) %}
+                {{ configuration.content }}
+            {% endfor %}
 
-            {% for location_var in served_domain.locations -%}
+
+            {% for location_var in served_domain.locations | default([]) -%}
             location {{location_var.condition}} {
                 {% if location_var.ignore_access is defined %}
                     {% for ip_range in served_domain.allowed_ip_ranges %}
                     allow {{ ip_range }};
                     {% endfor %}
                     deny all;
                 {% endif %}
-                
+
 
                 {{ location_var.content }}
             }
-
-
-
             {% endfor %}
+        }
     {% endfor %}
-    
-    }
-    {% if upstreams is defined %}
-    {% for upstream in upstreams %}
+
+    {% for upstream in upstreams | default([]) %}
```

Comment:
please prefix and document `global`, `upstreams`, `maps`.

Reply:
done